### PR TITLE
Refactor translator.ToURLMap to not re-fetch backend services

### PR DESCRIPTION
**Pull request by rramkumar1**
/assign @nicksardo @MrHohn 

Will update tests once approach is LGTM'd.

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -135,3 +136,17 @@ func trimFieldsEvenly(max int, fields ...string) []string {
 
 	return ret
 }
+
+// BackendRelativeResourcePath returns a relative path of the link for a
+// BackendService given its name.
+func BackendRelativeResourcePath(name string) string {
```
Comment (nicksardo) [inline pkg/utils/utils.go]: Let's use BackendService to not get confused with BackendService.Backend

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -135,3 +136,17 @@ func trimFieldsEvenly(max int, fields ...string) []string {
...
+	return fmt.Sprintf("global/backendServices/%v", name)
+}
+
+// BackendComparableGroupPath trims project and compute version from the SelfLink
+// for a global BackendService.
+// /global/backendServices/[BACKEND_SERVICE_NAME]
+func BackendComparableGroupPath(url string) string {
```
Comment (nicksardo) [inline pkg/utils/utils.go]: Group was referring to InstanceGroup, which isn't applicable for the BackendService.
Maybe "BackendServiceComparablePath"

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -135,3 +136,17 @@ func trimFieldsEvenly(max int, fields ...string) []string {
...
+	path_parts := strings.Split(url, "/global/")
+	return fmt.Sprintf("/global/%s", path_parts[1])
```
Comment (nicksardo) [inline pkg/utils/utils.go]: We should handle the error case of len(url) != 2... Guessing we return empty.

File: pkg/utils/utils.go (hunk 1 above)
Comment (rramkumar1) [reply pkg/utils/utils.go]: Done

File: pkg/utils/utils.go (hunk 2 above)
Comment (rramkumar1) [reply pkg/utils/utils.go]: Done

File: pkg/utils/utils.go (hunk 3 above)
Comment (rramkumar1) [reply pkg/utils/utils.go]: Done

**Comment by rramkumar1** [comment]
Adding tests now.

File: pkg/loadbalancers/l7.go [hunk 4]
```diff
@@ -824,7 +825,9 @@ func mapsEqual(a, b *compute.UrlMap) bool {
 					return false
 				}
 			}
-			if a.Service != b.Service {
+			// Trim down the url's for a.Service and b.Service to a comparable structure
+			// We do this because we update the UrlMap with relative links (not full) to backends.
+			if utils.BackendServiceComparablePath(a.Service) != utils.BackendServiceComparablePath(b.Service) {
```
Comment (nicksardo) [inline pkg/loadbalancers/l7.go]: There are two more lines with Service equality checks. 775 and 805.

File: pkg/utils/utils.go [hunk 5]
```diff
@@ -135,3 +136,20 @@ func trimFieldsEvenly(max int, fields ...string) []string {
...
+
+// BackendServiceRelativeResourcePath returns a relative path of the link for a
+// BackendService given its name.
+func BackendServiceRelativeResourcePath(name string) string {
+	return fmt.Sprintf("global/backendServices/%v", name)
+}
+
+// BackendServiceComparablePath trims project and compute version from the SelfLink
+// for a global BackendService.
+// /global/backendServices/[BACKEND_SERVICE_NAME]
```
Comment (nicksardo) [inline pkg/utils/utils.go]: Nit: leading slash

File: pkg/loadbalancers/l7.go (hunk 4 above)
Comment (rramkumar1) [reply pkg/loadbalancers/l7.go]: Done

File: pkg/utils/utils.go (hunk 5 above)
Comment (rramkumar1) [reply pkg/utils/utils.go]: Done

**Pull request merged**